data-prepper-api/src/test/java/org/opensearch/dataprepper/model/event/AggregateEventHandleTests.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.event;
+
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import org.junit.jupiter.api.Test;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.doAnswer;
+import org.mockito.Mock;
+
+import java.lang.ref.WeakReference;
+import java.time.Instant;
+
+class AggregateEventHandleTests {
+    @Mock
+    private AcknowledgementSet acknowledgementSet1;
+    @Mock
+    private AcknowledgementSet acknowledgementSet2;
+    private int count;
+    private int acquireCount;
+
+    @Test
+    void testBasic() {
+        Instant now = Instant.now();
+        AggregateEventHandle eventHandle = new AggregateEventHandle(now);
+        assertThat(eventHandle.getInternalOriginationTime(), equalTo(now));
+        assertThat(eventHandle.getExternalOriginationTime(), equalTo(null));
+        assertThat(eventHandle.hasAcknowledgementSet(), equalTo(false));
+        eventHandle.acquireReference();
+        eventHandle.release(true);
+    }
+
+    @Test
+    void testWithAcknowledgementSet() {
+        acquireCount = 0;
+        acknowledgementSet1 = mock(AcknowledgementSet.class);
+        acknowledgementSet2 = mock(AcknowledgementSet.class);
+        when(acknowledgementSet1.release(any(EventHandle.class), any(Boolean.class))).thenReturn(true);
+        when(acknowledgementSet2.release(any(EventHandle.class), any(Boolean.class))).thenReturn(true);
+        doAnswer(a -> {
+            acquireCount++;
+            return null;
+        }).when(acknowledgementSet1).acquire(any(EventHandle.class));
+        doAnswer(a -> {
+            acquireCount++;
+            return null;
+        }).when(acknowledgementSet2).acquire(any(EventHandle.class));
+        Instant now = Instant.now();
+        AggregateEventHandle eventHandle = new AggregateEventHandle(now);
+        assertThat(eventHandle.getInternalOriginationTime(), equalTo(now));
+        assertThat(eventHandle.getExternalOriginationTime(), equalTo(null));
+        eventHandle.addAcknowledgementSet(acknowledgementSet1);
+        // just do duplicate add
+        eventHandle.addAcknowledgementSet(acknowledgementSet1);
+        assertThat(eventHandle.hasAcknowledgementSet(), equalTo(true));
+        eventHandle.addAcknowledgementSet(acknowledgementSet2);
+        eventHandle.acquireReference();
+        verify(acknowledgementSet1).acquire(eventHandle);
+        verify(acknowledgementSet2).acquire(eventHandle);
+        assertThat(acquireCount, equalTo(2));
```

Comment:
Why do you need this assertion? And also, why the acquireCount tracking?

I believe that lines 67-68 fully assert the same thing.